src/aatoolbox/config/pathconfig.py

```diff
@@ -0,0 +1,22 @@
+"""Path configuration."""
+
+import os
+from dataclasses import dataclass
+from pathlib import Path
+
+BASE_DIR_ENV = "AA_DATA_DIR"
+
+
+@dataclass
+class PathConfig:
+    """Global directory parameters."""
+
+    public: str = "public"
+    private: str = "private"
+    raw: str = "raw"
+    processed: str = "processed"
+
+    def __init__(self, base_dir_env: str = BASE_DIR_ENV):
```

Comment:
So, had to do some reading about the `@dataclass` decorator as I'm not familiar, looks very cool. The only thing I notice is it seems it's recommended to use the `__post_init__` method, which becomes the last thing called by `__init__`. In this case, could you take advantage of the data class by changing this to:

```python
@dataclass
 class PathConfig:
     """Global directory parameters."""

     public: str = "public"
     private: str = "private"
     raw: str = "raw"
     processed: str = "processed"
     base_dir_env: str = BASE_DIR_ENV

     def __post_init__(self):
          self.base_path = Path(os.environ[self.base_dir_env])
```

Reply:
As usual your Googling skills reign supreme. Nice suggestion, fixed!